// ionic-selectable/__ivy_ngcc__/esm5/src/app/components/ionic-selectable/ionic-selectable-modal.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, HostBinding, HostListener, ViewChild } from '@angular/core';
import { IonContent, IonInfiniteScroll, IonSearchbar, NavParams } from '@ionic/angular';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@ionic/angular';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/forms';

var _c0 = ["searchbarComponent"];
function IonicSelectableModalComponent_ion_toolbar_1_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 12);
} if (rf & 2) {
    var ctx_r10 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r10.selectComponent.closeButtonTemplate);
} }
function IonicSelectableModalComponent_ion_toolbar_1_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r11 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r11.selectComponent.closeButtonText, " ");
} }
function IonicSelectableModalComponent_ion_toolbar_1_span_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 12);
} if (rf & 2) {
    var ctx_r12 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r12.selectComponent.titleTemplate);
} }
function IonicSelectableModalComponent_ion_toolbar_1_span_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r13 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r13.selectComponent.label, " ");
} }
function IonicSelectableModalComponent_ion_toolbar_1_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-toolbar", 9);
    ɵngcc0.ɵɵelementStart(1, "ion-buttons", 10);
    ɵngcc0.ɵɵelementStart(2, "ion-button", 11);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_toolbar_1_Template_ion_button_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.selectComponent._close(); });
    ɵngcc0.ɵɵtemplate(3, IonicSelectableModalComponent_ion_toolbar_1_span_3_Template, 1, 1, "span", 1);
    ɵngcc0.ɵɵtemplate(4, IonicSelectableModalComponent_ion_toolbar_1_span_4_Template, 2, 1, "span", 2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "ion-title");
    ɵngcc0.ɵɵtemplate(6, IonicSelectableModalComponent_ion_toolbar_1_span_6_Template, 1, 1, "span", 1);
    ɵngcc0.ɵɵtemplate(7, IonicSelectableModalComponent_ion_toolbar_1_span_7_Template, 2, 1, "span", 2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("color", ctx_r0.selectComponent.headerColor ? ctx_r0.selectComponent.headerColor : null);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("slot", ctx_r0.selectComponent.closeButtonSlot);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.selectComponent.closeButtonTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.selectComponent.closeButtonTemplate);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.selectComponent.titleTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.selectComponent.titleTemplate);
} }
function IonicSelectableModalComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 12);
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r1.selectComponent.headerTemplate);
} }
function IonicSelectableModalComponent_ion_toolbar_3_ion_searchbar_1_Template(rf, ctx) { if (rf & 1) {
    var _r20 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-searchbar", 15, 16);
    ɵngcc0.ɵɵlistener("ngModelChange", function IonicSelectableModalComponent_ion_toolbar_3_ion_searchbar_1_Template_ion_searchbar_ngModelChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r20); var ctx_r19 = ɵngcc0.ɵɵnextContext(2); return ctx_r19.selectComponent._searchText = $event; })("ionChange", function IonicSelectableModalComponent_ion_toolbar_3_ion_searchbar_1_Template_ion_searchbar_ionChange_0_listener() { ɵngcc0.ɵɵrestoreView(_r20); var ctx_r21 = ɵngcc0.ɵɵnextContext(2); return ctx_r21.selectComponent._filterItems(); })("ionClear", function IonicSelectableModalComponent_ion_toolbar_3_ion_searchbar_1_Template_ion_searchbar_ionClear_0_listener() { ɵngcc0.ɵɵrestoreView(_r20); var ctx_r22 = ɵngcc0.ɵɵnextContext(2); return ctx_r22.selectComponent._onSearchbarClear(); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r16 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngModel", ctx_r16.selectComponent._searchText)("placeholder", ctx_r16.selectComponent.searchPlaceholder)("debounce", ctx_r16.selectComponent.searchDebounce);
} }
function IonicSelectableModalComponent_ion_toolbar_3_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 17);
    ɵngcc0.ɵɵelement(1, "div", 12);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r17 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r17.selectComponent.messageTemplate);
} }
function IonicSelectableModalComponent_ion_toolbar_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-toolbar");
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_toolbar_3_ion_searchbar_1_Template, 2, 3, "ion-searchbar", 13);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_toolbar_3_div_2_Template, 2, 1, "div", 14);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r2.selectComponent.canSearch);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r2.selectComponent.messageTemplate);
} }
function IonicSelectableModalComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 18);
    ɵngcc0.ɵɵelement(1, "div", 19);
    ɵngcc0.ɵɵelement(2, "ion-spinner");
    ɵngcc0.ɵɵelementEnd();
} }
var _c1 = function (a0) { return { group: a0 }; };
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 26);
} if (rf & 2) {
    var group_r24 = ɵngcc0.ɵɵnextContext(2).$implicit;
    var ctx_r27 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r27.selectComponent.groupTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction1(2, _c1, group_r24));
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_ion_label_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-label");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var group_r24 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", group_r24.text, " ");
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 27);
    ɵngcc0.ɵɵelement(1, "div", 26);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var group_r24 = ɵngcc0.ɵɵnextContext(2).$implicit;
    var ctx_r29 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r29.selectComponent.groupEndTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction1(2, _c1, group_r24));
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-item-divider", 9);
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_span_1_Template, 1, 4, "span", 24);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_ion_label_2_Template, 2, 1, "ion-label", 2);
    ɵngcc0.ɵɵtemplate(3, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_div_3_Template, 2, 4, "div", 25);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r25 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("color", ctx_r25.selectComponent.groupColor ? ctx_r25.selectComponent.groupColor : null);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r25.selectComponent.groupTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r25.selectComponent.groupTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r25.selectComponent.groupEndTemplate);
} }
var _c2 = function (a0, a1) { return { item: a0, isItemSelected: a1 }; };
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 26);
} if (rf & 2) {
    var item_r33 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r34 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r34.selectComponent.itemTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction2(2, _c2, item_r33, ctx_r34.selectComponent._isItemSelected(item_r33)));
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_label_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-label");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r33 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r35 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r35.selectComponent._formatItem(item_r33), " ");
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 27);
    ɵngcc0.ɵɵelement(1, "div", 26);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r33 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r36 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r36.selectComponent.itemEndTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction2(2, _c2, item_r33, ctx_r36.selectComponent._isItemSelected(item_r33)));
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 26);
} if (rf & 2) {
    var item_r33 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r37 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r37.selectComponent.itemIconTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction2(2, _c2, item_r33, ctx_r37.selectComponent._isItemSelected(item_r33)));
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_icon_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "ion-icon", 31);
} if (rf & 2) {
    var item_r33 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r38 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("name", ctx_r38.selectComponent._isItemSelected(item_r33) ? "checkmark-circle" : "radio-button-off")("color", ctx_r38.selectComponent._isItemSelected(item_r33) ? "primary" : null)("slot", ctx_r38.selectComponent.itemIconSlot);
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_button_6_Template(rf, ctx) { if (rf & 1) {
    var _r47 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 32);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_button_6_Template_ion_button_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r47); var item_r33 = ɵngcc0.ɵɵnextContext().$implicit; var ctx_r46 = ɵngcc0.ɵɵnextContext(3); return ctx_r46.selectComponent._saveItem($event, item_r33); });
    ɵngcc0.ɵɵelement(1, "ion-icon", 33);
    ɵngcc0.ɵɵelementEnd();
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_button_7_Template(rf, ctx) { if (rf & 1) {
    var _r50 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 32);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_button_7_Template_ion_button_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r50); var item_r33 = ɵngcc0.ɵɵnextContext().$implicit; var ctx_r49 = ɵngcc0.ɵɵnextContext(3); return ctx_r49.selectComponent._deleteItemClick($event, item_r33); });
    ɵngcc0.ɵɵelement(1, "ion-icon", 34);
    ɵngcc0.ɵɵelementEnd();
} }
var _c3 = function (a0, a1) { return { "ionic-selectable-item-is-selected": a0, "ionic-selectable-item-is-disabled": a1 }; };
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_Template(rf, ctx) { if (rf & 1) {
    var _r53 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-item", 28);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_Template_ion_item_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r53); var item_r33 = ctx.$implicit; var ctx_r52 = ɵngcc0.ɵɵnextContext(3); return ctx_r52.selectComponent._select(item_r33); });
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_span_1_Template, 1, 5, "span", 24);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_label_2_Template, 2, 1, "ion-label", 2);
    ɵngcc0.ɵɵtemplate(3, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_div_3_Template, 2, 5, "div", 25);
    ɵngcc0.ɵɵtemplate(4, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_span_4_Template, 1, 5, "span", 24);
    ɵngcc0.ɵɵtemplate(5, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_icon_5_Template, 1, 3, "ion-icon", 29);
    ɵngcc0.ɵɵtemplate(6, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_button_6_Template, 2, 0, "ion-button", 30);
    ɵngcc0.ɵɵtemplate(7, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_ion_button_7_Template, 2, 0, "ion-button", 30);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r33 = ctx.$implicit;
    var ctx_r26 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(9, _c3, ctx_r26.selectComponent._isItemSelected(item_r33), ctx_r26.selectComponent._isItemDisabled(item_r33)))("disabled", ctx_r26.selectComponent._isItemDisabled(item_r33));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r26.selectComponent.itemTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r26.selectComponent.itemTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r26.selectComponent.itemEndTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r26.selectComponent.itemIconTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r26.selectComponent.itemIconTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r26.selectComponent.canSaveItem);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r26.selectComponent.canDeleteItem);
} }
function IonicSelectableModalComponent_ion_list_6_ion_item_group_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-item-group", 22);
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_divider_1_Template, 4, 4, "ion-item-divider", 0);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_ion_item_2_Template, 8, 12, "ion-item", 23);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var group_r24 = ctx.$implicit;
    var ctx_r23 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r23.selectComponent._hasGroups);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", group_r24.items);
} }
function IonicSelectableModalComponent_ion_list_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-list", 20);
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_list_6_ion_item_group_1_Template, 3, 2, "ion-item-group", 21);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r4.selectComponent._filteredGroups);
} }
function IonicSelectableModalComponent_div_7_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 12);
} if (rf & 2) {
    var ctx_r54 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r54.selectComponent.searchFailTemplate);
} }
function IonicSelectableModalComponent_div_7_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 36);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r55 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r55.selectComponent.searchFailText, " ");
} }
function IonicSelectableModalComponent_div_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_div_7_span_1_Template, 1, 1, "span", 1);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_div_7_div_2_Template, 2, 1, "div", 35);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r5.selectComponent.searchFailTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r5.selectComponent.searchFailTemplate);
} }
function IonicSelectableModalComponent_ion_infinite_scroll_8_Template(rf, ctx) { if (rf & 1) {
    var _r57 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-infinite-scroll", 37);
    ɵngcc0.ɵɵlistener("ionInfinite", function IonicSelectableModalComponent_ion_infinite_scroll_8_Template_ion_infinite_scroll_ionInfinite_0_listener() { ɵngcc0.ɵɵrestoreView(_r57); var ctx_r56 = ɵngcc0.ɵɵnextContext(); return ctx_r56.selectComponent._getMoreItems(); });
    ɵngcc0.ɵɵelement(1, "ion-infinite-scroll-content");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("disabled", !ctx_r6.selectComponent.hasInfiniteScroll);
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_divider_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-item-divider", 9);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var header_r60 = ctx.$implicit;
    var ctx_r58 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("color", ctx_r58.selectComponent.groupColor ? ctx_r58.selectComponent.groupColor : null);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", header_r60, " ");
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_span_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 26);
} if (rf & 2) {
    var item_r61 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r62 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r62.selectComponent.itemTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction2(2, _c2, item_r61, ctx_r62.selectComponent._isItemSelected(item_r61)));
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_label_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-label");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r61 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r63 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r63.selectComponent._formatItem(item_r61), " ");
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 27);
    ɵngcc0.ɵɵelement(1, "div", 26);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r61 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r64 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r64.selectComponent.itemEndTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction2(2, _c2, item_r61, ctx_r64.selectComponent._isItemSelected(item_r61)));
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 26);
} if (rf & 2) {
    var item_r61 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r65 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r65.selectComponent.itemIconTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction2(2, _c2, item_r61, ctx_r65.selectComponent._isItemSelected(item_r61)));
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_icon_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "ion-icon", 31);
} if (rf & 2) {
    var item_r61 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r66 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("name", ctx_r66.selectComponent._isItemSelected(item_r61) ? "checkmark-circle" : "radio-button-off")("color", ctx_r66.selectComponent._isItemSelected(item_r61) ? "primary" : null)("slot", ctx_r66.selectComponent.itemIconSlot);
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_button_6_Template(rf, ctx) { if (rf & 1) {
    var _r75 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 32);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_button_6_Template_ion_button_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r75); var item_r61 = ɵngcc0.ɵɵnextContext().$implicit; var ctx_r74 = ɵngcc0.ɵɵnextContext(2); return ctx_r74.selectComponent._saveItem($event, item_r61); });
    ɵngcc0.ɵɵelement(1, "ion-icon", 33);
    ɵngcc0.ɵɵelementEnd();
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_button_7_Template(rf, ctx) { if (rf & 1) {
    var _r78 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 32);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_button_7_Template_ion_button_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r78); var item_r61 = ɵngcc0.ɵɵnextContext().$implicit; var ctx_r77 = ɵngcc0.ɵɵnextContext(2); return ctx_r77.selectComponent._deleteItemClick($event, item_r61); });
    ɵngcc0.ɵɵelement(1, "ion-icon", 34);
    ɵngcc0.ɵɵelementEnd();
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_Template(rf, ctx) { if (rf & 1) {
    var _r81 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-item", 28);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_Template_ion_item_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r81); var item_r61 = ctx.$implicit; var ctx_r80 = ɵngcc0.ɵɵnextContext(2); return ctx_r80.selectComponent._select(item_r61); });
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_span_1_Template, 1, 5, "span", 24);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_label_2_Template, 2, 1, "ion-label", 2);
    ɵngcc0.ɵɵtemplate(3, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_div_3_Template, 2, 5, "div", 25);
    ɵngcc0.ɵɵtemplate(4, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_span_4_Template, 1, 5, "span", 24);
    ɵngcc0.ɵɵtemplate(5, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_icon_5_Template, 1, 3, "ion-icon", 29);
    ɵngcc0.ɵɵtemplate(6, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_button_6_Template, 2, 0, "ion-button", 30);
    ɵngcc0.ɵɵtemplate(7, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_ion_button_7_Template, 2, 0, "ion-button", 30);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r61 = ctx.$implicit;
    var ctx_r59 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(9, _c3, ctx_r59.selectComponent._isItemSelected(item_r61), ctx_r59.selectComponent._isItemDisabled(item_r61)))("disabled", ctx_r59.selectComponent._isItemDisabled(item_r61));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r59.selectComponent.itemTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r59.selectComponent.itemTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r59.selectComponent.itemEndTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r59.selectComponent.itemIconTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r59.selectComponent.itemIconTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r59.selectComponent.canSaveItem);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r59.selectComponent.canDeleteItem);
} }
function IonicSelectableModalComponent_ion_virtual_scroll_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-virtual-scroll", 38);
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_divider_1_Template, 2, 2, "ion-item-divider", 39);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_virtual_scroll_9_ion_item_2_Template, 8, 12, "ion-item", 40);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r7 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("items", ctx_r7.selectComponent._filteredGroups[0].items)("headerFn", ctx_r7.selectComponent.virtualScrollHeaderFn)("approxItemHeight", ctx_r7.selectComponent.virtualScrollApproxItemHeight);
} }
var _c4 = function (a0) { return { "top.px": a0 }; };
var _c5 = function (a0) { return { "height": a0 }; };
var _c6 = function (a0, a1) { return { item: a0, isAdd: a1 }; };
function IonicSelectableModalComponent_div_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 41);
    ɵngcc0.ɵɵelementStart(1, "div", 42);
    ɵngcc0.ɵɵelement(2, "span", 26);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r8 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(4, _c4, ctx_r8._header.offsetHeight));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(6, _c5, ctx_r8.selectComponent._addItemTemplateFooterHeight));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r8.selectComponent.addItemTemplate)("ngTemplateOutletContext", ɵngcc0.ɵɵpureFunction2(8, _c6, ctx_r8.selectComponent._itemToAdd, ctx_r8.selectComponent._itemToAdd === null));
} }
function IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_2_Template(rf, ctx) { if (rf & 1) {
    var _r88 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-col");
    ɵngcc0.ɵɵelementStart(1, "ion-button", 44);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_2_Template_ion_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r88); var ctx_r87 = ɵngcc0.ɵɵnextContext(3); return ctx_r87.selectComponent._clear(); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r84 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("disabled", !ctx_r84.selectComponent._selectedItems.length);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r84.selectComponent.clearButtonText, " ");
} }
function IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_3_Template(rf, ctx) { if (rf & 1) {
    var _r90 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-col");
    ɵngcc0.ɵɵelementStart(1, "ion-button", 45);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_3_Template_ion_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r90); var ctx_r89 = ɵngcc0.ɵɵnextContext(3); return ctx_r89.selectComponent._addItemClick(); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r85 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r85.selectComponent.addButtonText, " ");
} }
function IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_4_Template(rf, ctx) { if (rf & 1) {
    var _r92 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-col");
    ɵngcc0.ɵɵelementStart(1, "ion-button", 44);
    ɵngcc0.ɵɵlistener("click", function IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_4_Template_ion_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r92); var ctx_r91 = ɵngcc0.ɵɵnextContext(3); return ctx_r91.selectComponent._confirm(); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r86 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("disabled", !ctx_r86.selectComponent.isConfirmButtonEnabled);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r86.selectComponent.confirmButtonText, " ");
} }
function IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-toolbar");
    ɵngcc0.ɵɵelementStart(1, "ion-row");
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_2_Template, 3, 2, "ion-col", 2);
    ɵngcc0.ɵɵtemplate(3, IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_3_Template, 3, 1, "ion-col", 2);
    ɵngcc0.ɵɵtemplate(4, IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_ion_col_4_Template, 3, 2, "ion-col", 2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r82 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r82.selectComponent.canClear);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r82.selectComponent.canAddItem);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r82.selectComponent.isMultiple || ctx_r82.selectComponent.hasConfirmButton);
} }
function IonicSelectableModalComponent_ion_footer_11_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 12);
} if (rf & 2) {
    var ctx_r83 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r83.selectComponent.footerTemplate);
} }
var _c7 = function (a0) { return { "visibility": a0 }; };
function IonicSelectableModalComponent_ion_footer_11_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-footer", 43);
    ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_footer_11_ion_toolbar_1_Template, 5, 3, "ion-toolbar", 2);
    ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_ion_footer_11_div_2_Template, 1, 1, "div", 1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r9 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c7, ctx_r9.selectComponent._isFooterVisible ? "initial" : "hidden"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r9.selectComponent.footerTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r9.selectComponent.footerTemplate);
} }
var IonicSelectableModalComponent = /** @class */ (function () {
    function IonicSelectableModalComponent(navParams, _element) {
        var _this = this;
        this.navParams = navParams;
        this._element = _element;
        this._cssClass = true;
        this.selectComponent = this.navParams.get('selectComponent');
        this.selectComponent._modalComponent = this;
        this.selectComponent._selectedItems = [];
        if (!this.selectComponent._isNullOrWhiteSpace(this.selectComponent.value)) {
            if (this.selectComponent.isMultiple) {
                this.selectComponent.value.forEach((/**
                 * @param {?} item
                 * @return {?}
                 */
                function (item) {
                    _this.selectComponent._selectedItems.push(item);
                }));
            }
            else {
                this.selectComponent._selectedItems.push(this.selectComponent.value);
            }
        }
        this.selectComponent._setItemsToConfirm(this.selectComponent._selectedItems);
    }
    Object.defineProperty(IonicSelectableModalComponent.prototype, "_canClearCssClass", {
        get: /**
         * @return {?}
         */
        function () {
            return this.selectComponent.canClear;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IonicSelectableModalComponent.prototype, "_isMultipleCssClass", {
        get: /**
         * @return {?}
         */
        function () {
            return this.selectComponent.isMultiple;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IonicSelectableModalComponent.prototype, "_isSearchingCssClass", {
        get: /**
         * @return {?}
         */
        function () {
            return this.selectComponent._isSearching;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IonicSelectableModalComponent.prototype, "_isIos", {
        get: /**
         * @return {?}
         */
        function () {
            return this.selectComponent._isIos;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    IonicSelectableModalComponent.prototype._isMD = /**
     * @return {?}
     */
    function () {
        return this.selectComponent._isMD;
    };
    Object.defineProperty(IonicSelectableModalComponent.prototype, "_isAddItemTemplateVisibleCssClass", {
        get: /**
         * @return {?}
         */
        function () {
            return this.selectComponent._isAddItemTemplateVisible;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    IonicSelectableModalComponent.prototype.onResize = /**
     * @return {?}
     */
    function () {
        // ion-footer inside the template might change its height when
        // device orientation changes.
        this.selectComponent._positionAddItemTemplate();
    };
    /**
     * @return {?}
     */
    IonicSelectableModalComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this._header = this._element.nativeElement.querySelector('ion-header');
        if (this._searchbarComponent && this.selectComponent.shouldFocusSearchbar) {
            // Focus after a delay because focus doesn't work without it.
            setTimeout((/**
             * @return {?}
             */
            function () {
                _this._searchbarComponent.setFocus();
            }), 1000);
        }
    };
    /** @nocollapse */
    IonicSelectableModalComponent.ctorParameters = function () { return [
        { type: NavParams },
        { type: ElementRef }
    ]; };
    IonicSelectableModalComponent.propDecorators = {
        _content: [{ type: ViewChild, args: [IonContent,] }],
        _searchbarComponent: [{ type: ViewChild, args: ['searchbarComponent',] }],
        _infiniteScroll: [{ type: ViewChild, args: [IonInfiniteScroll,] }],
        _cssClass: [{ type: HostBinding, args: ['class.ionic-selectable-modal',] }],
        _canClearCssClass: [{ type: HostBinding, args: ['class.ionic-selectable-modal-can-clear',] }],
        _isMultipleCssClass: [{ type: HostBinding, args: ['class.ionic-selectable-modal-is-multiple',] }],
        _isSearchingCssClass: [{ type: HostBinding, args: ['class.ionic-selectable-modal-is-searching',] }],
        _isIos: [{ type: HostBinding, args: ['class.ionic-selectable-modal-ios',] }],
        _isMD: [{ type: HostBinding, args: ['class.ionic-selectable-modal-md',] }],
        _isAddItemTemplateVisibleCssClass: [{ type: HostBinding, args: ['class.ionic-selectable-modal-is-add-item-template-visible',] }],
        onResize: [{ type: HostListener, args: ['window:resize',] }]
    };
IonicSelectableModalComponent.ɵfac = function IonicSelectableModalComponent_Factory(t) { return new (t || IonicSelectableModalComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NavParams), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
IonicSelectableModalComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IonicSelectableModalComponent, selectors: [["ionic-selectable-modal"]], viewQuery: function IonicSelectableModalComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(IonContent, true);
        ɵngcc0.ɵɵviewQuery(_c0, true);
        ɵngcc0.ɵɵviewQuery(IonInfiniteScroll, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx._content = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx._searchbarComponent = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx._infiniteScroll = _t.first);
    } }, hostVars: 14, hostBindings: function IonicSelectableModalComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("resize", function IonicSelectableModalComponent_resize_HostBindingHandler() { return ctx.onResize(); }, false, ɵngcc0.ɵɵresolveWindow);
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("ionic-selectable-modal", ctx._cssClass)("ionic-selectable-modal-can-clear", ctx._canClearCssClass)("ionic-selectable-modal-is-multiple", ctx._isMultipleCssClass)("ionic-selectable-modal-is-searching", ctx._isSearchingCssClass)("ionic-selectable-modal-ios", ctx._isIos)("ionic-selectable-modal-md", ctx._isMD)("ionic-selectable-modal-is-add-item-template-visible", ctx._isAddItemTemplateVisibleCssClass);
    } }, decls: 12, vars: 10, consts: [[3, "color", 4, "ngIf"], [3, "ngTemplateOutlet", 4, "ngIf"], [4, "ngIf"], ["class", "ionic-selectable-spinner", 4, "ngIf"], ["class", "ion-no-margin", 4, "ngIf"], [3, "disabled", "ionInfinite", 4, "ngIf"], ["class", "ion-no-margin", 3, "items", "headerFn", "approxItemHeight", 4, "ngIf"], ["class", "ionic-selectable-add-item-template", 3, "ngStyle", 4, "ngIf"], [3, "ngStyle", 4, "ngIf"], [3, "color"], [3, "slot"], [3, "click"], [3, "ngTemplateOutlet"], [3, "ngModel", "placeholder", "debounce", "ngModelChange", "ionChange", "ionClear", 4, "ngIf"], ["class", "ionic-selectable-message", 4, "ngIf"], [3, "ngModel", "placeholder", "debounce", "ngModelChange", "ionChange", "ionClear"], ["searchbarComponent", ""], [1, "ionic-selectable-message"], [1, "ionic-selectable-spinner"], [1, "ionic-selectable-spinner-background"], [1, "ion-no-margin"], ["class", "ionic-selectable-group", 4, "ngFor", "ngForOf"], [1, "ionic-selectable-group"], ["button", "true", "detail", "false", "class", "ionic-selectable-item", 3, "ngClass", "disabled", "click", 4, "ngFor", "ngForOf"], [3, "ngTemplateOutlet", "ngTemplateOutletContext", 4, "ngIf"], ["slot", "end", 4, "ngIf"], [3, "ngTemplateOutlet", "ngTemplateOutletContext"], ["slot", "end"], ["button", "true", "detail", "false", 1, "ionic-selectable-item", 3, "ngClass", "disabled", "click"], [3, "name", "color", "slot", 4, "ngIf"], ["class", "ionic-selectable-item-button", "slot", "end", "fill", "outline", 3, "click", 4, "ngIf"], [3, "name", "color", "slot"], ["slot", "end", "fill", "outline", 1, "ionic-selectable-item-button", 3, "click"], ["slot", "icon-only", "name", "md-create"], ["slot", "icon-only", "name", "md-trash"], ["class", "ion-margin", 4, "ngIf"], [1, "ion-margin"], [3, "disabled", "ionInfinite"], [1, "ion-no-margin", 3, "items", "headerFn", "approxItemHeight"], [3, "color", 4, "virtualHeader"], ["button", "true", "detail", "false", "class", "ionic-selectable-item", 3, "ngClass", "disabled", "click", 4, "virtualItem"], [1, "ionic-selectable-add-item-template", 3, "ngStyle"], [1, "ionic-selectable-add-item-template-inner", 3, "ngStyle"], [3, "ngStyle"], ["expand", "full", 3, "disabled", "click"], ["expand", "full", 3, "click"]], template: function IonicSelectableModalComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "ion-header");
        ɵngcc0.ɵɵtemplate(1, IonicSelectableModalComponent_ion_toolbar_1_Template, 8, 6, "ion-toolbar", 0);
        ɵngcc0.ɵɵtemplate(2, IonicSelectableModalComponent_div_2_Template, 1, 1, "div", 1);
        ɵngcc0.ɵɵtemplate(3, IonicSelectableModalComponent_ion_toolbar_3_Template, 3, 2, "ion-toolbar", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "ion-content");
        ɵngcc0.ɵɵtemplate(5, IonicSelectableModalComponent_div_5_Template, 3, 0, "div", 3);
        ɵngcc0.ɵɵtemplate(6, IonicSelectableModalComponent_ion_list_6_Template, 2, 1, "ion-list", 4);
        ɵngcc0.ɵɵtemplate(7, IonicSelectableModalComponent_div_7_Template, 3, 2, "div", 2);
        ɵngcc0.ɵɵtemplate(8, IonicSelectableModalComponent_ion_infinite_scroll_8_Template, 2, 1, "ion-infinite-scroll", 5);
        ɵngcc0.ɵɵtemplate(9, IonicSelectableModalComponent_ion_virtual_scroll_9_Template, 3, 3, "ion-virtual-scroll", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(10, IonicSelectableModalComponent_div_10_Template, 3, 11, "div", 7);
        ɵngcc0.ɵɵtemplate(11, IonicSelectableModalComponent_ion_footer_11_Template, 3, 5, "ion-footer", 8);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.selectComponent.headerTemplate);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selectComponent.headerTemplate);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selectComponent.canSearch || ctx.selectComponent.messageTemplate);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selectComponent._isSearching);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.selectComponent.hasVirtualScroll && ctx.selectComponent._hasFilteredItems);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.selectComponent._hasFilteredItems);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.selectComponent.hasVirtualScroll);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selectComponent.hasVirtualScroll && ctx.selectComponent._hasFilteredItems);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selectComponent._isAddItemTemplateVisible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.selectComponent._footerButtonsCount > 0 || ctx.selectComponent.footerTemplate);
    } }, directives: [ɵngcc1.IonHeader, ɵngcc2.NgIf, ɵngcc1.IonContent, ɵngcc1.IonToolbar, ɵngcc1.IonButtons, ɵngcc1.IonButton, ɵngcc1.IonTitle, ɵngcc2.NgTemplateOutlet, ɵngcc1.IonSearchbar, ɵngcc1.TextValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel, ɵngcc1.IonSpinner, ɵngcc1.IonList, ɵngcc2.NgForOf, ɵngcc1.IonItemGroup, ɵngcc1.IonItemDivider, ɵngcc1.IonLabel, ɵngcc1.IonItem, ɵngcc2.NgClass, ɵngcc1.IonIcon, ɵngcc1.IonInfiniteScroll, ɵngcc1.IonInfiniteScrollContent, ɵngcc1.IonVirtualScroll, ɵngcc1.VirtualHeader, ɵngcc1.VirtualItem, ɵngcc2.NgStyle, ɵngcc1.IonFooter, ɵngcc1.IonRow, ɵngcc1.IonCol], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IonicSelectableModalComponent, [{
        type: Component,
        args: [{
                selector: 'ionic-selectable-modal',
                template: "<ion-header>\n  <ion-toolbar *ngIf=\"!selectComponent.headerTemplate\"\n    [color]=\"selectComponent.headerColor ? selectComponent.headerColor : null\">\n    <ion-buttons [slot]=\"selectComponent.closeButtonSlot\">\n      <ion-button (click)=\"selectComponent._close()\">\n        <span *ngIf=\"selectComponent.closeButtonTemplate\"\n          [ngTemplateOutlet]=\"selectComponent.closeButtonTemplate\">\n        </span>\n        <span *ngIf=\"!selectComponent.closeButtonTemplate\">\n          {{selectComponent.closeButtonText}}\n        </span>\n      </ion-button>\n    </ion-buttons>\n    <ion-title>\n      <!-- Need span for for text ellipsis. -->\n      <span *ngIf=\"selectComponent.titleTemplate\"\n        [ngTemplateOutlet]=\"selectComponent.titleTemplate\">\n      </span>\n      <span *ngIf=\"!selectComponent.titleTemplate\">\n        {{selectComponent.label}}\n      </span>\n    </ion-title>\n  </ion-toolbar>\n  <div *ngIf=\"selectComponent.headerTemplate\"\n    [ngTemplateOutlet]=\"selectComponent.headerTemplate\">\n  </div>\n  <ion-toolbar\n    *ngIf=\"selectComponent.canSearch || selectComponent.messageTemplate\">\n    <ion-searchbar *ngIf=\"selectComponent.canSearch\" #searchbarComponent\n      [(ngModel)]=\"selectComponent._searchText\"\n      (ionChange)=\"selectComponent._filterItems()\"\n      (ionClear)=\"selectComponent._onSearchbarClear()\"\n      [placeholder]=\"selectComponent.searchPlaceholder\"\n      [debounce]=\"selectComponent.searchDebounce\">\n    </ion-searchbar>\n    <div class=\"ionic-selectable-message\"\n      *ngIf=\"selectComponent.messageTemplate\">\n      <div [ngTemplateOutlet]=\"selectComponent.messageTemplate\">\n      </div>\n    </div>\n  </ion-toolbar>\n</ion-header>\n<ion-content>\n  <div class=\"ionic-selectable-spinner\" *ngIf=\"selectComponent._isSearching\">\n    <div class=\"ionic-selectable-spinner-background\"></div>\n    <ion-spinner></ion-spinner>\n  </div>\n  <ion-list class=\"ion-no-margin\"\n    *ngIf=\"!selectComponent.hasVirtualScroll && selectComponent._hasFilteredItems\">\n    <ion-item-group *ngFor=\"let group of selectComponent._filteredGroups\"\n      class=\"ionic-selectable-group\">\n      <ion-item-divider *ngIf=\"selectComponent._hasGroups\"\n        [color]=\"selectComponent.groupColor ? selectComponent.groupColor : null\">\n        <!-- Need span for for text ellipsis. -->\n        <span *ngIf=\"selectComponent.groupTemplate\"\n          [ngTemplateOutlet]=\"selectComponent.groupTemplate\"\n          [ngTemplateOutletContext]=\"{ group: group }\">\n        </span>\n        <!-- Need ion-label for text ellipsis. -->\n        <ion-label *ngIf=\"!selectComponent.groupTemplate\">\n          {{group.text}}\n        </ion-label>\n        <div *ngIf=\"selectComponent.groupEndTemplate\" slot=\"end\">\n          <div [ngTemplateOutlet]=\"selectComponent.groupEndTemplate\"\n            [ngTemplateOutletContext]=\"{ group: group }\">\n          </div>\n        </div>\n      </ion-item-divider>\n      <ion-item button=\"true\" detail=\"false\" *ngFor=\"let item of group.items\"\n        (click)=\"selectComponent._select(item)\" class=\"ionic-selectable-item\"\n        [ngClass]=\"{\n          'ionic-selectable-item-is-selected': selectComponent._isItemSelected(item),\n          'ionic-selectable-item-is-disabled': selectComponent._isItemDisabled(item)\n        }\" [disabled]=\"selectComponent._isItemDisabled(item)\">\n        <!-- Need span for text ellipsis. -->\n        <span *ngIf=\"selectComponent.itemTemplate\"\n          [ngTemplateOutlet]=\"selectComponent.itemTemplate\"\n          [ngTemplateOutletContext]=\"{ item: item, isItemSelected: selectComponent._isItemSelected(item) }\">\n        </span>\n        <!-- Need ion-label for text ellipsis. -->\n        <ion-label *ngIf=\"!selectComponent.itemTemplate\">\n          {{selectComponent._formatItem(item)}}\n        </ion-label>\n        <div *ngIf=\"selectComponent.itemEndTemplate\" slot=\"end\">\n          <div [ngTemplateOutlet]=\"selectComponent.itemEndTemplate\"\n            [ngTemplateOutletContext]=\"{ item: item, isItemSelected: selectComponent._isItemSelected(item) }\">\n          </div>\n        </div>\n        <span *ngIf=\"selectComponent.itemIconTemplate\"\n          [ngTemplateOutlet]=\"selectComponent.itemIconTemplate\"\n          [ngTemplateOutletContext]=\"{ item: item, isItemSelected: selectComponent._isItemSelected(item) }\">\n        </span>\n        <ion-icon *ngIf=\"!selectComponent.itemIconTemplate\"\n          [name]=\"selectComponent._isItemSelected(item) ? 'checkmark-circle' : 'radio-button-off'\"\n          [color]=\"selectComponent._isItemSelected(item) ? 'primary' : null\"\n          [slot]=\"selectComponent.itemIconSlot\">\n        </ion-icon>\n        <ion-button *ngIf=\"selectComponent.canSaveItem\"\n          class=\"ionic-selectable-item-button\" slot=\"end\" fill=\"outline\"\n          (click)=\"selectComponent._saveItem($event, item)\">\n          <ion-icon slot=\"icon-only\" name=\"md-create\"></ion-icon>\n        </ion-button>\n        <ion-button *ngIf=\"selectComponent.canDeleteItem\"\n          class=\"ionic-selectable-item-button\" slot=\"end\" fill=\"outline\"\n          (click)=\"selectComponent._deleteItemClick($event, item)\">\n          <ion-icon slot=\"icon-only\" name=\"md-trash\"></ion-icon>\n        </ion-button>\n      </ion-item>\n    </ion-item-group>\n  </ion-list>\n  <!-- Fail text should be above InfiniteScroll to avoid a gap when no items are found. -->\n  <div *ngIf=\"!selectComponent._hasFilteredItems\">\n    <span *ngIf=\"selectComponent.searchFailTemplate\"\n      [ngTemplateOutlet]=\"selectComponent.searchFailTemplate\">\n    </span>\n    <div *ngIf=\"!selectComponent.searchFailTemplate\" class=\"ion-margin\">\n      {{selectComponent.searchFailText}}\n    </div>\n  </div>\n  <ion-infinite-scroll *ngIf=\"!selectComponent.hasVirtualScroll\"\n    [disabled]=\"!selectComponent.hasInfiniteScroll\"\n    (ionInfinite)=\"selectComponent._getMoreItems()\">\n    <ion-infinite-scroll-content></ion-infinite-scroll-content>\n  </ion-infinite-scroll>\n  <ion-virtual-scroll class=\"ion-no-margin\"\n    *ngIf=\"selectComponent.hasVirtualScroll && selectComponent._hasFilteredItems\"\n    [items]=\"selectComponent._filteredGroups[0].items\"\n    [headerFn]=\"selectComponent.virtualScrollHeaderFn\"\n    [approxItemHeight]=\"selectComponent.virtualScrollApproxItemHeight\">\n    <ion-item-divider *virtualHeader=\"let header\"\n      [color]=\"selectComponent.groupColor ? selectComponent.groupColor : null\">\n      {{header}}\n    </ion-item-divider>\n    <ion-item button=\"true\" detail=\"false\" *virtualItem=\"let item\"\n      (click)=\"selectComponent._select(item)\" class=\"ionic-selectable-item\"\n      [ngClass]=\"{\n        'ionic-selectable-item-is-selected': selectComponent._isItemSelected(item),\n        'ionic-selectable-item-is-disabled': selectComponent._isItemDisabled(item)\n      }\" [disabled]=\"selectComponent._isItemDisabled(item)\">\n      <!-- Need span for text ellipsis. -->\n      <span *ngIf=\"selectComponent.itemTemplate\"\n        [ngTemplateOutlet]=\"selectComponent.itemTemplate\"\n        [ngTemplateOutletContext]=\"{ item: item, isItemSelected: selectComponent._isItemSelected(item) }\">\n      </span>\n      <!-- Need ion-label for text ellipsis. -->\n      <ion-label *ngIf=\"!selectComponent.itemTemplate\">\n        {{selectComponent._formatItem(item)}}\n      </ion-label>\n      <div *ngIf=\"selectComponent.itemEndTemplate\" slot=\"end\">\n        <div [ngTemplateOutlet]=\"selectComponent.itemEndTemplate\"\n          [ngTemplateOutletContext]=\"{ item: item, isItemSelected: selectComponent._isItemSelected(item) }\">\n        </div>\n      </div>\n      <span *ngIf=\"selectComponent.itemIconTemplate\"\n        [ngTemplateOutlet]=\"selectComponent.itemIconTemplate\"\n        [ngTemplateOutletContext]=\"{ item: item, isItemSelected: selectComponent._isItemSelected(item) }\">\n      </span>\n      <ion-icon *ngIf=\"!selectComponent.itemIconTemplate\"\n        [name]=\"selectComponent._isItemSelected(item) ? 'checkmark-circle' : 'radio-button-off'\"\n        [color]=\"selectComponent._isItemSelected(item) ? 'primary' : null\"\n        [slot]=\"selectComponent.itemIconSlot\">\n      </ion-icon>\n      <ion-button *ngIf=\"selectComponent.canSaveItem\"\n        class=\"ionic-selectable-item-button\" slot=\"end\" fill=\"outline\"\n        (click)=\"selectComponent._saveItem($event, item)\">\n        <ion-icon slot=\"icon-only\" name=\"md-create\"></ion-icon>\n      </ion-button>\n      <ion-button *ngIf=\"selectComponent.canDeleteItem\"\n        class=\"ionic-selectable-item-button\" slot=\"end\" fill=\"outline\"\n        (click)=\"selectComponent._deleteItemClick($event, item)\">\n        <ion-icon slot=\"icon-only\" name=\"md-trash\"></ion-icon>\n      </ion-button>\n    </ion-item>\n  </ion-virtual-scroll>\n</ion-content>\n<div class=\"ionic-selectable-add-item-template\"\n  *ngIf=\"selectComponent._isAddItemTemplateVisible\"\n  [ngStyle]=\"{ 'top.px': _header.offsetHeight }\">\n  <div class=\"ionic-selectable-add-item-template-inner\"\n    [ngStyle]=\"{ 'height': selectComponent._addItemTemplateFooterHeight }\">\n    <span [ngTemplateOutlet]=\"selectComponent.addItemTemplate\"\n      [ngTemplateOutletContext]=\"{ item: selectComponent._itemToAdd, isAdd: selectComponent._itemToAdd === null }\">\n    </span>\n  </div>\n</div>\n<ion-footer\n  *ngIf=\"selectComponent._footerButtonsCount > 0 || selectComponent.footerTemplate\"\n  [ngStyle]=\"{ 'visibility': selectComponent._isFooterVisible ? 'initial' : 'hidden' }\">\n  <ion-toolbar *ngIf=\"!selectComponent.footerTemplate\">\n    <ion-row>\n      <ion-col *ngIf=\"selectComponent.canClear\">\n        <ion-button expand=\"full\" (click)=\"selectComponent._clear()\"\n          [disabled]=\"!selectComponent._selectedItems.length\">\n          {{selectComponent.clearButtonText}}\n        </ion-button>\n      </ion-col>\n      <ion-col *ngIf=\"selectComponent.canAddItem\">\n        <ion-button expand=\"full\" (click)=\"selectComponent._addItemClick()\">\n          {{selectComponent.addButtonText}}\n        </ion-button>\n      </ion-col>\n      <ion-col\n        *ngIf=\"selectComponent.isMultiple || selectComponent.hasConfirmButton\">\n        <ion-button expand=\"full\" (click)=\"selectComponent._confirm()\"\n          [disabled]=\"!selectComponent.isConfirmButtonEnabled\">\n          {{selectComponent.confirmButtonText}}\n        </ion-button>\n      </ion-col>\n    </ion-row>\n  </ion-toolbar>\n  <div *ngIf=\"selectComponent.footerTemplate\"\n    [ngTemplateOutlet]=\"selectComponent.footerTemplate\">\n  </div>\n</ion-footer>\n"
            }]
    }], function () { return [{ type: ɵngcc1.NavParams }, { type: ɵngcc0.ElementRef }]; }, { _cssClass: [{
            type: HostBinding,
            args: ['class.ionic-selectable-modal']
        }], _canClearCssClass: [{
            type: HostBinding,
            args: ['class.ionic-selectable-modal-can-clear']
        }], _isMultipleCssClass: [{
            type: HostBinding,
            args: ['class.ionic-selectable-modal-is-multiple']
        }], _isSearchingCssClass: [{
            type: HostBinding,
            args: ['class.ionic-selectable-modal-is-searching']
        }], _isIos: [{
            type: HostBinding,
            args: ['class.ionic-selectable-modal-ios']
        }], _isMD: [{
            type: HostBinding,
            args: ['class.ionic-selectable-modal-md']
        }], _isAddItemTemplateVisibleCssClass: [{
            type: HostBinding,
            args: ['class.ionic-selectable-modal-is-add-item-template-visible']
        }], onResize: [{
            type: HostListener,
            args: ['window:resize']
        }], _content: [{
            type: ViewChild,
            args: [IonContent]
        }], _searchbarComponent: [{
            type: ViewChild,
            args: ['searchbarComponent']
        }], _infiniteScroll: [{
            type: ViewChild,
            args: [IonInfiniteScroll]
        }] }); })();
    return IonicSelectableModalComponent;
}());
export { IonicSelectableModalComponent };
if (false) {
    /** @type {?} */
    IonicSelectableModalComponent.prototype._content;
    /** @type {?} */
    IonicSelectableModalComponent.prototype._header;
    /** @type {?} */
    IonicSelectableModalComponent.prototype.selectComponent;
    /** @type {?} */
    IonicSelectableModalComponent.prototype._searchbarComponent;
    /** @type {?} */
    IonicSelectableModalComponent.prototype._infiniteScroll;
    /** @type {?} */
    IonicSelectableModalComponent.prototype._cssClass;
    /**
     * @type {?}
     * @private
     */
    IonicSelectableModalComponent.prototype.navParams;
    /** @type {?} */
    IonicSelectableModalComponent.prototype._element;
}
