// angular4-events/__ivy_ngcc__/esm/src/angular4-events.module.js
import * as ɵngcc0 from '@angular/core';
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { EventsService } from './angular4-events.service';
import { NgModule } from '@angular/core';
var EventsModule = /** @class */ (function () {
    function EventsModule() {
    }
    EventsModule_1 = EventsModule;
    EventsModule.forRoot = function () {
        return {
            ngModule: EventsModule_1,
            providers: [
                EventsService
            ]
        };
    };
    var EventsModule_1;
EventsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: EventsModule });
EventsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function EventsModule_Factory(t) { return new (t || EventsModule)(); }, providers: [
        EventsService
    ] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EventsModule, [{
        type: NgModule,
        args: [{
                providers: [
                    EventsService
                ]
            }]
    }], function () { return []; }, null); })();
    return EventsModule;
}());
export { EventsModule };
