// @syncfusion/ej2-angular-pdfviewer/__ivy_ngcc__/@syncfusion/ej2-angular-pdfviewer.es5.js
import * as ɵngcc0 from '@angular/core';
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { ChangeDetectionStrategy, Component, ElementRef, Injector, NgModule, Renderer2, ViewContainerRef } from '@angular/core';
import { ComponentBase, ComponentMixins, setValue } from '@syncfusion/ej2-angular-base';
import { Annotation, BookmarkView, FormFields, LinkAnnotation, Magnification, Navigation, PdfViewer, Print, TextSearch, TextSelection, ThumbnailView, Toolbar } from '@syncfusion/ej2-pdfviewer';
import { CommonModule } from '@angular/common';
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
var inputs = ['ajaxRequestSettings', 'annotationSelectorSettings', 'annotationSettings', 'annotationToolbarSettings', 'annotations', 'areaSettings', 'arrowSettings', 'circleSettings', 'contextMenuOption', 'customStampItems', 'customStampSettings', 'disableContextMenuItems', 'distanceSettings', 'documentPath', 'downloadFileName', 'drawingObject', 'enableAnnotation', 'enableAnnotationToolbar', 'enableAutoComplete', 'enableBookmark', 'enableCommentPanel', 'enableDownload', 'enableFormFields', 'enableFormFieldsValidation', 'enableFreeText', 'enableHandwrittenSignature', 'enableHyperlink', 'enableImportAnnotationMeasurement', 'enableInkAnnotation', 'enableMagnification', 'enableMeasureAnnotation', 'enableMultiLineOverlap', 'enableMultiPageAnnotation', 'enableNavigation', 'enableNavigationToolbar', 'enablePersistence', 'enablePinchZoom', 'enablePrint', 'enableRtl', 'enableShapeAnnotation', 'enableShapeLabel', 'enableStampAnnotations', 'enableStickyNotesAnnotation', 'enableTextMarkupAnnotation', 'enableTextMarkupResizer', 'enableTextSearch', 'enableTextSelection', 'enableThumbnail', 'enableToolbar', 'enableZoomOptimization', 'formFieldCollections', 'freeTextSettings', 'handWrittenSignatureSettings', 'height', 'highlightSettings', 'hyperlinkOpenState', 'inkAnnotationSettings', 'interactionMode', 'isCommandPanelOpen', 'isExtractText', 'isFormFieldsDocument', 'isSignatureEditable', 'isThumbnailViewOpen', 'lineSettings', 'locale', 'measurementSettings', 'perimeterSettings', 'polygonSettings', 'printMode', 'radiusSettings', 'rectangleSettings', 'restrictZoomRequest', 'retryCount', 'scrollSettings', 'selectedItems', 'serverActionSettings', 'serviceUrl', 'shapeLabelSettings', 'showNotificationDialog', 'stampItemSettings', 'stampSettings', 'stickyNotesSettings', 'strikethroughSettings', 'textSearchColorSettings', 'tileRenderingSettings', 'toolbarSettings', 'underlineSettings', 'volumeSettings', 'width', 'zoomMode', 'zoomValue'];
var outputs = ['addSignature', 'ajaxRequestFailed', 'ajaxRequestInitiate', 'annotationAdd', 'annotationDoubleClick', 'annotationMouseover', 'annotationMove', 'annotationPropertiesChange', 'annotationRemove', 'annotationResize', 'annotationSelect', 'documentLoad', 'documentLoadFailed', 'documentUnload', 'downloadEnd', 'downloadStart', 'exportFailed', 'exportStart', 'exportSuccess', 'extractTextCompleted', 'hyperlinkClick', 'hyperlinkMouseOver', 'importFailed', 'importStart', 'importSuccess', 'moveSignature', 'pageChange', 'pageClick', 'pageMouseover', 'printEnd', 'printStart', 'removeSignature', 'resizeSignature', 'signaturePropertiesChange', 'signatureSelect', 'textSearchComplete', 'textSearchHighlight', 'textSearchStart', 'textSelectionEnd', 'textSelectionStart', 'thumbnailClick', 'validateFormFields', 'zoomChange'];
var twoWays = [];
/**
 * `ejs-pdfviewer` represents the Angular PdfViewer Component.
 * ```html
 * <ejs-pdfviewer></ejs-pdfviewer>
 * ```
 */
var PdfViewerComponent = /** @class */ (function (_super) {
    __extends(PdfViewerComponent, _super);
    /**
     * @param {?} ngEle
     * @param {?} srenderer
     * @param {?} viewContainerRef
     * @param {?} injector
     */
    function PdfViewerComponent(ngEle, srenderer, viewContainerRef, injector) {
        var _this = _super.call(this) || this;
        _this.ngEle = ngEle;
        _this.srenderer = srenderer;
        _this.viewContainerRef = viewContainerRef;
        _this.injector = injector;
        _this.element = _this.ngEle.nativeElement;
        _this.injectedModules = _this.injectedModules || [];
        try {
            var mod = _this.injector.get('PdfViewerLinkAnnotation');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_a) { }
        try {
            var mod = _this.injector.get('PdfViewerBookmarkView');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_b) { }
        try {
            var mod = _this.injector.get('PdfViewerMagnification');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_c) { }
        try {
            var mod = _this.injector.get('PdfViewerThumbnailView');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_d) { }
        try {
            var mod = _this.injector.get('PdfViewerToolbar');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_e) { }
        try {
            var mod = _this.injector.get('PdfViewerNavigation');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_f) { }
        try {
            var mod = _this.injector.get('PdfViewerPrint');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_g) { }
        try {
            var mod = _this.injector.get('PdfViewerTextSelection');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_h) { }
        try {
            var mod = _this.injector.get('PdfViewerTextSearch');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_j) { }
        try {
            var mod = _this.injector.get('PdfViewerAnnotation');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_k) { }
        try {
            var mod = _this.injector.get('PdfViewerFormFields');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_l) { }
        _this.registerEvents(outputs);
        _this.addTwoWay.call(_this, twoWays);
        setValue('currentInstance', _this, _this.viewContainerRef);
        _this.context = new ComponentBase();
        return _this;
    }
    /**
     * @return {?}
     */
    PdfViewerComponent.prototype.ngOnInit = function () {
        this.context.ngOnInit(this);
    };
    /**
     * @return {?}
     */
    PdfViewerComponent.prototype.ngAfterViewInit = function () {
        this.context.ngAfterViewInit(this);
    };
    /**
     * @return {?}
     */
    PdfViewerComponent.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy(this);
    };
    /**
     * @return {?}
     */
    PdfViewerComponent.prototype.ngAfterContentChecked = function () {
        this.context.ngAfterContentChecked(this);
    };
PdfViewerComponent.ɵfac = function PdfViewerComponent_Factory(t) { return new (t || PdfViewerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Injector)); };
PdfViewerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PdfViewerComponent, selectors: [["ejs-pdfviewer"]], inputs: { ajaxRequestSettings: "ajaxRequestSettings", annotationSelectorSettings: "annotationSelectorSettings", annotationSettings: "annotationSettings", annotationToolbarSettings: "annotationToolbarSettings", annotations: "annotations", areaSettings: "areaSettings", arrowSettings: "arrowSettings", circleSettings: "circleSettings", contextMenuOption: "contextMenuOption", customStampItems: "customStampItems", customStampSettings: "customStampSettings", disableContextMenuItems: "disableContextMenuItems", distanceSettings: "distanceSettings", documentPath: "documentPath", downloadFileName: "downloadFileName", drawingObject: "drawingObject", enableAnnotation: "enableAnnotation", enableAnnotationToolbar: "enableAnnotationToolbar", enableAutoComplete: "enableAutoComplete", enableBookmark: "enableBookmark", enableCommentPanel: "enableCommentPanel", enableDownload: "enableDownload", enableFormFields: "enableFormFields", enableFormFieldsValidation: "enableFormFieldsValidation", enableFreeText: "enableFreeText", enableHandwrittenSignature: "enableHandwrittenSignature", enableHyperlink: "enableHyperlink", enableImportAnnotationMeasurement: "enableImportAnnotationMeasurement", enableInkAnnotation: "enableInkAnnotation", enableMagnification: "enableMagnification", enableMeasureAnnotation: "enableMeasureAnnotation", enableMultiLineOverlap: "enableMultiLineOverlap", enableMultiPageAnnotation: "enableMultiPageAnnotation", enableNavigation: "enableNavigation", enableNavigationToolbar: "enableNavigationToolbar", enablePersistence: "enablePersistence", enablePinchZoom: "enablePinchZoom", enablePrint: "enablePrint", enableRtl: "enableRtl", enableShapeAnnotation: "enableShapeAnnotation", enableShapeLabel: "enableShapeLabel", enableStampAnnotations: "enableStampAnnotations", enableStickyNotesAnnotation: "enableStickyNotesAnnotation", enableTextMarkupAnnotation: "enableTextMarkupAnnotation", enableTextMarkupResizer: "enableTextMarkupResizer", enableTextSearch: "enableTextSearch", enableTextSelection: "enableTextSelection", enableThumbnail: "enableThumbnail", enableToolbar: "enableToolbar", enableZoomOptimization: "enableZoomOptimization", formFieldCollections: "formFieldCollections", freeTextSettings: "freeTextSettings", handWrittenSignatureSettings: "handWrittenSignatureSettings", height: "height", highlightSettings: "highlightSettings", hyperlinkOpenState: "hyperlinkOpenState", inkAnnotationSettings: "inkAnnotationSettings", interactionMode: "interactionMode", isCommandPanelOpen: "isCommandPanelOpen", isExtractText: "isExtractText", isFormFieldsDocument: "isFormFieldsDocument", isSignatureEditable: "isSignatureEditable", isThumbnailViewOpen: "isThumbnailViewOpen", lineSettings: "lineSettings", locale: "locale", measurementSettings: "measurementSettings", perimeterSettings: "perimeterSettings", polygonSettings: "polygonSettings", printMode: "printMode", radiusSettings: "radiusSettings", rectangleSettings: "rectangleSettings", restrictZoomRequest: "restrictZoomRequest", retryCount: "retryCount", scrollSettings: "scrollSettings", selectedItems: "selectedItems", serverActionSettings: "serverActionSettings", serviceUrl: "serviceUrl", shapeLabelSettings: "shapeLabelSettings", showNotificationDialog: "showNotificationDialog", stampItemSettings: "stampItemSettings", stampSettings: "stampSettings", stickyNotesSettings: "stickyNotesSettings", strikethroughSettings: "strikethroughSettings", textSearchColorSettings: "textSearchColorSettings", tileRenderingSettings: "tileRenderingSettings", toolbarSettings: "toolbarSettings", underlineSettings: "underlineSettings", volumeSettings: "volumeSettings", width: "width", zoomMode: "zoomMode", zoomValue: "zoomValue" }, outputs: { addSignature: "addSignature", ajaxRequestFailed: "ajaxRequestFailed", ajaxRequestInitiate: "ajaxRequestInitiate", annotationAdd: "annotationAdd", annotationDoubleClick: "annotationDoubleClick", annotationMouseover: "annotationMouseover", annotationMove: "annotationMove", annotationPropertiesChange: "annotationPropertiesChange", annotationRemove: "annotationRemove", annotationResize: "annotationResize", annotationSelect: "annotationSelect", documentLoad: "documentLoad", documentLoadFailed: "documentLoadFailed", documentUnload: "documentUnload", downloadEnd: "downloadEnd", downloadStart: "downloadStart", exportFailed: "exportFailed", exportStart: "exportStart", exportSuccess: "exportSuccess", extractTextCompleted: "extractTextCompleted", hyperlinkClick: "hyperlinkClick", hyperlinkMouseOver: "hyperlinkMouseOver", importFailed: "importFailed", importStart: "importStart", importSuccess: "importSuccess", moveSignature: "moveSignature", pageChange: "pageChange", pageClick: "pageClick", pageMouseover: "pageMouseover", printEnd: "printEnd", printStart: "printStart", removeSignature: "removeSignature", resizeSignature: "resizeSignature", signaturePropertiesChange: "signaturePropertiesChange", signatureSelect: "signatureSelect", textSearchComplete: "textSearchComplete", textSearchHighlight: "textSearchHighlight", textSearchStart: "textSearchStart", textSelectionEnd: "textSelectionEnd", textSelectionStart: "textSelectionStart", thumbnailClick: "thumbnailClick", validateFormFields: "validateFormFields", zoomChange: "zoomChange" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 0, vars: 0, template: function PdfViewerComponent_Template(rf, ctx) { }, encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PdfViewerComponent, [{
        type: Component,
        args: [{
                selector: 'ejs-pdfviewer',
                inputs: inputs,
                outputs: outputs,
                template: '',
                changeDetection: ChangeDetectionStrategy.OnPush,
                queries: {}
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ViewContainerRef }, { type: ɵngcc0.Injector }]; }, null); })();
    return PdfViewerComponent;
}(PdfViewer));
/**
 * @nocollapse
 */
PdfViewerComponent.ctorParameters = function () { return [
    { type: ElementRef, },
    { type: Renderer2, },
    { type: ViewContainerRef, },
    { type: Injector, },
]; };
PdfViewerComponent = __decorate([
    ComponentMixins([ComponentBase]),
    __metadata("design:paramtypes", [ElementRef,
        Renderer2,
        ViewContainerRef,
        Injector])
], PdfViewerComponent);
/**
 * NgModule definition for the PdfViewer component.
 */
var PdfViewerModule = /** @class */ (function () {
    function PdfViewerModule() {
    }
PdfViewerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: PdfViewerModule });
PdfViewerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function PdfViewerModule_Factory(t) { return new (t || PdfViewerModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(PdfViewerModule, { declarations: function () { return [PdfViewerComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [PdfViewerComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PdfViewerModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [
                    PdfViewerComponent
                ],
                exports: [
                    PdfViewerComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return PdfViewerModule;
}());
/**
 * @nocollapse
 */
PdfViewerModule.ctorParameters = function () { return []; };
var LinkAnnotationService = { provide: 'PdfViewerLinkAnnotation', useValue: LinkAnnotation };
var BookmarkViewService = { provide: 'PdfViewerBookmarkView', useValue: BookmarkView };
var MagnificationService = { provide: 'PdfViewerMagnification', useValue: Magnification };
var ThumbnailViewService = { provide: 'PdfViewerThumbnailView', useValue: ThumbnailView };
var ToolbarService = { provide: 'PdfViewerToolbar', useValue: Toolbar };
var NavigationService = { provide: 'PdfViewerNavigation', useValue: Navigation };
var PrintService = { provide: 'PdfViewerPrint', useValue: Print };
var TextSelectionService = { provide: 'PdfViewerTextSelection', useValue: TextSelection };
var TextSearchService = { provide: 'PdfViewerTextSearch', useValue: TextSearch };
var AnnotationService = { provide: 'PdfViewerAnnotation', useValue: Annotation };
var FormFieldsService = { provide: 'PdfViewerFormFields', useValue: FormFields };
/**
 * NgModule definition for the PdfViewer component with providers.
 */
var PdfViewerAllModule = /** @class */ (function () {
    function PdfViewerAllModule() {
    }
PdfViewerAllModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: PdfViewerAllModule });
PdfViewerAllModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function PdfViewerAllModule_Factory(t) { return new (t || PdfViewerAllModule)(); }, providers: [
        LinkAnnotationService,
        BookmarkViewService,
        MagnificationService,
        ThumbnailViewService,
        ToolbarService,
        NavigationService,
        PrintService,
        TextSelectionService,
        TextSearchService,
        AnnotationService,
        FormFieldsService
    ], imports: [[CommonModule, PdfViewerModule],
        PdfViewerModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(PdfViewerAllModule, { imports: function () { return [CommonModule,
        PdfViewerModule]; }, exports: function () { return [PdfViewerModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PdfViewerAllModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PdfViewerModule],
                exports: [
                    PdfViewerModule
                ],
                providers: [
                    LinkAnnotationService,
                    BookmarkViewService,
                    MagnificationService,
                    ThumbnailViewService,
                    ToolbarService,
                    NavigationService,
                    PrintService,
                    TextSelectionService,
                    TextSearchService,
                    AnnotationService,
                    FormFieldsService
                ]
            }]
    }], function () { return []; }, null); })();
    return PdfViewerAllModule;
}());
/**
 * @nocollapse
 */
PdfViewerAllModule.ctorParameters = function () { return []; };
/**
 * Generated bundle index. Do not edit.
 */
export { PdfViewerComponent, PdfViewerModule, PdfViewerAllModule, LinkAnnotationService, BookmarkViewService, MagnificationService, ThumbnailViewService, ToolbarService, NavigationService, PrintService, TextSelectionService, TextSearchService, AnnotationService, FormFieldsService, inputs as ɵa, outputs as ɵb };
export { Annotation, LinkAnnotation, TextMarkupAnnotation, MeasureAnnotation, ShapeAnnotation, StampAnnotation, StickyNotesAnnotation, FreeTextAnnotation, InputElement, InkAnnotation, NavigationPane, PdfViewerBase, TextLayer, ContextMenu, FontStyle, ContextMenuItem, AnnotationResizerLocation, CursorType, DynamicStampItem, SignStampItem, StandardBusinessStampItem, AjaxHandler, Signature, Magnification, Navigation, ThumbnailView, Toolbar, AnnotationToolbar, ToolbarSettings, AjaxRequestSettings, CustomStampItem, AnnotationToolbarSettings, ServerActionSettings, StrikethroughSettings, UnderlineSettings, HighlightSettings, LineSettings, ArrowSettings, RectangleSettings, CircleSettings, ShapeLabelSettings, PolygonSettings, StampSettings, CustomStampSettings, DistanceSettings, PerimeterSettings, AreaSettings, RadiusSettings, VolumeSettings, InkAnnotationSettings, StickyNotesSettings, MeasurementSettings, FreeTextSettings, AnnotationSelectorSettings, TextSearchColorSettings, HandWrittenSignatureSettings, AnnotationSettings, DocumentTextCollectionSettings, TextDataSettings, RectangleBounds, TileRenderingSettings, ScrollSettings, FormFieldCollections, StampItemSettings, PdfViewer, BookmarkView, TextSelection, TextSearch, Print, FormFields, Drawing, findActiveElement, findObjectsUnderMouse, findObjectUnderMouse, CalculateLeaderPoints, findElementUnderMouse, insertObject, findTargetShapeElement, findObjects, findActivePage, ActiveElements, getConnectorPoints, getSegmentPath, updateSegmentElement, getSegmentElement, updateDecoratorElement, getDecoratorElement, clipDecorators, clipDecorator, initDistanceLabel, updateDistanceLabel, updateRadiusLabel, initPerimeterLabel, updatePerimeterLabel, removePerimeterLabel, updateCalibrateLabel, getPolygonPath, textElement, initLeaders, initLeader, isPointOverConnector, findNearestPoint, getDecoratorShape, renderAdornerLayer, createSvg, isLineShapes, setElementStype, findPointsLength, findPerimeterLength, getBaseShapeAttributes, getFunction, cloneObject, cloneArray, getInternalProperties, isLeader, PdfBounds, PdfFont, PdfAnnotationBase, ZOrderPageTable, Selector, ToolBase, SelectTool, MoveTool, StampTool, InkDrawingTool, ConnectTool, ResizeTool, NodeDrawingTool, PolygonDrawingTool, LineTool, RotateTool } from '@syncfusion/ej2-pdfviewer';
